fhttp/http_test.go

```diff
@@ -968,6 +968,30 @@ func TestAddHTTPS(t *testing.T) {
 	}
 }
 
+func TestValidateAndAddBasicAuthentication(t *testing.T) {
+	var tests = []struct {
+		o                  HTTPOptions
+		isCredentialsValid bool
+		isAuthHeaderAdded  bool
+	}{
+		{HTTPOptions{UserCredentials: "foo:foo"}, false, true},
```

Comment:
this should be true (valid credentials) and true (auth header added)

or your variable isCredentialsValid is wrong

Reply:
I should have written that isCredentialsNotValid. But, I'm keeping variable name same and changing boolean values to their not